salt/_modules/vault.py

```diff
@@ -0,0 +1,858 @@
+                    log.debug('Policy "%s" has been updated.', policy["name"])
+                else:
+                    new_policies.append(policy["name"])
+                    log.debug('Policy "%s" has been created.', policy["name"])
+
+            # Build return object
+            ret['changes']['old'] = remote_policies
+            ret['changes']['new'] = new_policies or "No changes"
+        except Exception:
+            raise
+
+        log.info('Finished pushing policies local config folder to vault.')
+
+    def cleanup_policies(self, client, remote_policies, local_policies, ret):
+        """Removes policies that are not present in the local config
+
+        Arguments:
+            client {hvac} -- hvac client
+            remote_policies {list} -- policies current on the remote vault server
+            local_policies {list} --policies from local config
+            ret {dict} -- salt state result
+        """
+        log.info('Cleaning up vault policies...')
+        has_change = False
+        try:
+            for policy in remote_policies:
+                if policy not in [pol['name'] for pol in local_policies]:
+                    log.debug(
+                        '"%s" is not found in configs folder. Removing it from vault...', policy)
+                    has_change = True
+                    client.sys.delete_policy(name=policy)
+                    log.debug('"%s" is removed.', policy)
+
+            if has_change:
+                ret['changes']['new'] = [ob['name'] for ob in local_policies]
+        except Exception:
+            raise
+
+        log.info('Finished cleaning up vault policies.')
+
+
+class VaultAuthManager():
+    """Module for managing Vault Authentication Methods
+    """
+
+    def __init__(self):
+      """Initialize Authentication Manager
+      """
+      log.info("Initializing Vault Auth Manager...")
+
+    def get_remote_auth_methods(self, client, ret):
+        """Retrieve authentication methods from remote vault server
+
+        Arguments:
+            client {hvac} -- hvac client
+            ret {dict} -- result from state
+
+        Returns:
+            list -- auth methods
+        """
+        log.info('Retrieving auth methods from Vault...')
+        auth_resp = client.sys.list_auth_methods()
+
+        auth_methods = []
+        try:
+            for auth_method in auth_resp['data']:
+                auth_methods.append(
+                    VaultAuthMethod(
+                        type=auth_resp[auth_method]['type'],
+                        path=(auth_resp[auth_method]["path"]
+                              if 'path' in auth_resp[auth_method] else auth_method),
+                        description=auth_resp[auth_method]["description"],
+                        config=OrderedDict(
+                            sorted(auth_resp[auth_method]["config"].items()))
+                    )
+                )
+        except Exception:
```

Comment:
Since we are not actually handling any exception, just re-raising, this try/except block is not needed. Can remove it and let things raise exceptions normally.